From: Hannes Reinecke <hare@suse.de>
Date: Thu, 4 Sep 2014 10:18:50 +0200
Subject: scsi_dh_alua: disable ALUA handling for non-disk devices
References: bnc#876633
Patch-Mainline: n/a

Switch to blacklisting all non-disk devices during
initializing the ALUA device handler.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/device_handler/scsi_dh_alua.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/scsi/device_handler/scsi_dh_alua.c b/drivers/scsi/device_handler/scsi_dh_alua.c
index 54e60e7..985ea50 100644
--- a/drivers/scsi/device_handler/scsi_dh_alua.c
+++ b/drivers/scsi/device_handler/scsi_dh_alua.c
@@ -344,12 +344,12 @@ static int alua_check_tpgs(struct scsi_device *sdev, struct alua_dh_data *h)
 			    ALUA_DH_NAME);
 		return SCSI_DH_DEV_UNSUPP;
 	}
-	if (sdev->type == TYPE_RAID ||
-	    sdev->type == TYPE_ENCLOSURE ||
-	    sdev->type == 0x1F) {
+	if (sdev->type != TYPE_DISK &&
+	    sdev->type != TYPE_RBC &&
+	    sdev->type != TYPE_OSD) {
 		h->tpgs = TPGS_MODE_NONE;
 		sdev_printk(KERN_INFO, sdev,
-			    "%s: disable for enclosure devices\n",
+			    "%s: disable for non-disk devices\n",
 			    ALUA_DH_NAME);
 		return SCSI_DH_DEV_UNSUPP;
 	}
-- 
1.8.5.2

